api/lib/domain/validators/superior-schooling-registration-set-validator.js

```diff
@@ -0,0 +1,36 @@
+const Joi = require('@hapi/joi').extend(require('@hapi/joi-date'));
+const { EntityValidationError } = require('../errors');
+
+const validationConfiguration = { abortEarly: false, allowUnknown: true };
+
+const validationSchema = Joi.object({
+  withStudentNumber: Joi.array().unique('studentNumber'),
+  withoutStudentNumber: Joi.array()
+    .unique((a, b) => a.lastName === b.lastName && a.firstName === b.firstName && a.birthdate.getTime() === b.birthdate.getTime())
+});
+
+module.exports = {
+  checkValidation(superiorSchoolingRegistrationSet) {
+    const registrationsToValidate = {
+      withStudentNumber: [],
+      withoutStudentNumber: []
+    };
+
+    superiorSchoolingRegistrationSet.registrations().forEach((registration) => {
+      if (registration.studentNumber) {
+        registrationsToValidate.withStudentNumber.push(registration);
+      }
+      else
+      {
+        registrationsToValidate.withoutStudentNumber.push(registration);
+      }
```

Comment:
indentation un peu bizarre
```suggestion
      if (registration.studentNumber) {
        registrationsToValidate.withStudentNumber.push(registration);
      } else {
        registrationsToValidate.withoutStudentNumber.push(registration);
      }
```

Reply:
Done